src/pages/Dashboard.tsx

```diff
@@ -176,6 +154,50 @@ const Dashboard = () => {
         };
     };
 
+    const [isTaskModelOpen, setIsTaskModelOpen] = useState<boolean>(false);
+    const handleOpenTaskModal = () => setIsTaskModelOpen(true);
+    const handleCloseTaskModal = () => setIsTaskModelOpen(false);
```

Comment:
you can do `handleToggleTaskModal = setIsTaskModalOpen((prev) => !prev)` if you think that'd work or be easier. Also Model should be Modal